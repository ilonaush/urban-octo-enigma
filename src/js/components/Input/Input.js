import React from 'react';
import PropTypes from 'prop-types';

const Input = ({placeholder, onChange, value, name, className, ...props}) => {
  return (
      <input className={className}
             placeholder={placeholder}
             onChange={onChange}
             value={value}
             name={name}
             {...props}
      />
  );
}

Input.propTypes = {
  value: PropTypes.string
};
Input.defaultProps = {
  value: '',
  onChange: () => {
  }
};


export default Input;
