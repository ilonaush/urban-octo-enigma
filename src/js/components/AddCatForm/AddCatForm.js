import React, {Component} from 'react';
import "./AddCatForm.styl";
import Input from "../Input/Input";
import PropTypes from 'prop-types';

export default class AddCatForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
            cat: {
                name: '',
                age: '',
                color: '',
                reason: '',
                location: '',
                health: ''
            }
        };
    }

    /**
     * handles input change
     * @param value
     * @param name
     */
    handleChange = ({target: {value, name}}) => {
        this.setState((prevState) => ({
            cat: {
                ...prevState.cat,
                [name]: value
            }
        }))
    };


    render() {
        const {cat} = this.state;
        const {handleSubmit} = this.props;
        return (
            <form className='addWorkerForm' onSubmit={(e) => handleSubmit(e, cat)}>
                <h3>Please share some info about the cat with us</h3>
                <Input id='name-input'
                       name='name'
                       placeholder='Name'
                       onChange={this.handleChange}
                       value={cat.name}
                       required
                        />
                <Input name='age'
                       placeholder='Age'
                       onChange={this.handleChange}
                       value={cat.age}
                       required
                />
                <Input name='color'
                       placeholder='Color'
                       onChange={this.handleChange}
                       value={cat.color}
                       required
                />
                <textarea name='reason'
                       placeholder='Reason'
                       onChange={this.handleChange}
                       value={cat.reason}
                />
                <Input name='location'
                       placeholder='Location'
                       onChange={this.handleChange}
                       value={cat.location}
                       required
                />
                <select
                    value={cat.health}
                    name='health'
                    onChange={this.handleChange}>
                        <option value="Satisfying">Satisfying</option>
                        <option value="Minor problems">Minor problems</option>
                        <option value="Not satisfying">Not satisfying</option>
                </select>
                <button id='submit-btn' type='submit'>Accept the cat</button>
            </form>
        );
    }
}

AddCatForm.propTypes = {
    handleSubmit: PropTypes.func,
};

