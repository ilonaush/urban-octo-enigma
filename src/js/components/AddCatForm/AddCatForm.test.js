import React from 'react';
import AddCatForm from './AddCatForm';
import {mount} from "enzyme/build";
import Input from "../Input/Input";

describe('AddCatForm', function () {

  beforeEach(() => {
    this.wrapper = mount(<AddCatForm/>)
  });

  it('renders correctly', () => {
    expect(this.wrapper).toMatchSnapshot();
  });

  it('calls submit function  when  the form is submitted', () => {
    const handleSubmit = jest.fn();

    this.wrapper.setProps({handleSubmit});

    const button = this.wrapper.find('#submit-btn');
    button.simulate('submit');
    expect(handleSubmit).toHaveBeenCalledTimes(1);
    expect(handleSubmit).toHaveBeenCalled();
  });

  it('change worker state when  the form is submitted', () => {
    const firstInput = this.wrapper.find(Input).at(0);
    expect(this.wrapper.state().cat.name).toEqual('');
    firstInput.simulate('change', {target: {value: 'mike', name: 'name'}});
    expect(this.wrapper.state().cat.name).toEqual('mike');
  });
});






