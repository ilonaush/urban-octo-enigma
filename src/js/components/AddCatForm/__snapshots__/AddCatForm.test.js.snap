// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCatForm renders correctly 1`] = `
<form
  className="addWorkerForm"
  onSubmit={[Function]}
>
  <h3>
    Please share some info about the cat with us
  </h3>
  <input
    id="name-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    required={true}
    value=""
  />
  <input
    name="age"
    onChange={[Function]}
    placeholder="Age"
    required={true}
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    placeholder="Color"
    required={true}
    value=""
  />
  <textarea
    name="reason"
    onChange={[Function]}
    placeholder="Reason"
    value=""
  />
  <input
    name="location"
    onChange={[Function]}
    placeholder="Location"
    required={true}
    value=""
  />
  <select
    name="health"
    onChange={[Function]}
    value=""
  >
    <option
      value="Satisfying"
    >
      Satisfying
    </option>
    <option
      value="Minor problems"
    >
      Minor problems
    </option>
    <option
      value="Not satisfying"
    >
      Not satisfying
    </option>
  </select>
  <button
    id="submit-btn"
    type="submit"
  >
    Accept the cat
  </button>
</form>
`;
